pkgs/development/tools/profiling/heaptrack/default.nix

```diff
@@ -17,10 +17,14 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake extra-cmake-modules ];
   buildInputs = [
-    zlib boost libunwind elfutils sparsehash
+    zlib boost libunwind elfutils sparsehash makeWrapper
     qtbase kio kitemmodels threadweaver kconfigwidgets kcoreaddons kdiagram
   ];
 
+  postInstall = ''
+    wrapProgram $out/bin/heaptrack_gui --set QT_PLUGIN_PATH ${qtbase}/${qtbase.qtPluginPrefix}
```

Comment:
Have you tried to install `heaptrack_gui` in your nix profile?

Reply:
The impurity for qt is necessary because the way qt searches for plugins.

Reply:
Indeed, it works just fine when impurely installed and run through a user profile.  However, with this change it also works when package is used outside of a `nix-env`.

Reply:
Setting `QT_PLUGIN_PATH` will break integration with desktop environments: https://github.com/NixOS/nixpkgs/issues/24256#issuecomment-328349797

This needs to be fixed in a central place.

Reply:
This seems a such a fix: https://github.com/NixOS/nixpkgs/pull/44047